### DISCUSSION TITLE
🐛 Selecting a saved search with spaces in the name triggers an error on landing page.

### DISCUSSION THREAD

**Discussion by schalkneethling** [community/community#63708] [Projects and Issues]
### Select Topic Area

Bug

### Body

I recently discovered the saved search feature. 🎉 I saved a search with the name "Vuejs Good First Issues." When I open the search field, it shows my saved search as a suggestion (`saved:"Vuejs Good First Issues"`). However, when I click on it, I get an error on the landing page.

> Possible unrecognized qualifier, searching for this term literally saved:"Vuejs Good First Issues"

It also shows this message twice in the UI. Is this because there are spaces in the name, or whether the name was supposed to be expanded into the search string? The docs suggest that that is the case.

> Clicking on a saved search entry will add the query to the search bar and filter the suggestions accordingly.

https://docs.github.com/en/search-github/github-code-search/using-github-code-search#creating-and-managing-saved-searches

**Comment by schalkneethling** [comment]
> ### Select Topic Area
> Bug
> 
> ### Body
> I recently discovered the saved search feature. 🎉 I saved a search with the name "Vuejs Good First Issues." When I open the search field, it shows my saved search as a suggestion (`saved:"Vuejs Good First Issues"`). However, when I click on it, I get an error on the landing page.
> 
> > Possible unrecognized qualifier, searching for this term literally saved:"Vuejs Good First Issues"
> 
> It also shows this message twice in the UI. Is this because there are spaces in the name, or whether the name was supposed to be expanded into the search string? The docs suggest that that is the case.
> 
> > Clicking on a saved search entry will add the query to the search bar and filter the suggestions accordingly.
> 
> https://docs.github.com/en/search-github/github-code-search/using-github-code-search#creating-and-managing-saved-searches

I edited the search to not have spaces in the name with the same end result when selecting the saved search in the UI.

**Comment by rmoyet1029** [comment]
Seems like instead of adding the saved query to the search bar it just adds saved:{someName} to the search bar. Anyone else experiencing this?

**Comment by sangram-prodigal** [comment]
facing the same issue. Is this being picked up for resolution any time soon?

**Reply by sangram-prodigal** [reply]
actually, this DOES work. I have a saved search where I look for all open PRs within a specific org that I belong to which have been marked to me as "review requested". When I search using the entire search string
```
org:some-org-name type:pr review-requested:my-user-name is:open
```
it works *as expected* meaning it understands I am looking for PRs and hence selects the "Pull Requests" tab from the side nav bar when the search results appear. When I instead use the "saved:my-saved-search" search, the search runs correctly (with this full query) but the side nav bar doesn't go to "Pull Requests" and remains instead on the default "Code" space. That is what I was seeing as the error page. 

Putting this here since it might help others who have raised this. 

@rmoyet1029 @schalkneethling is your issue similar, or still persists / is different?

**Comment by kimdre** [comment]
I also experience this issue. The saved search does not seem to get converted to the actual search query and instead it just searches for the exact name of the saved query like `saved:"Some Name" `.

**Comment by kimdre** [comment]
Github support confirmed me this issue and told me that they are already investigating this internally.

**Comment by LiteBrite82** [comment]
💬 Your Product Feedback Has Been Submitted 🎉
Thank you for taking the time to share your insights with us! Your feedback is invaluable as we build a better GitHub experience for all our users.

Here's what you can expect moving forward ⏩
- Your input will be carefully reviewed and cataloged by members of our product teams. 
- Other users may engage with your post, sharing their own perspectives or experiences. 
- GitHub staff may reach out for further clarification or insight. 

Where to look to see what's shipping 👀
- Read the [Changelog](https://github.blog/changelog/) for real-time updates on the latest GitHub features, enhancements, and calls for feedback.
- Explore our [Product Roadmap](https://github.com/orgs/github/projects/4247), which details upcoming major releases and initiatives.

What you can do in the meantime 💻
- Upvote and comment on other user feedback Discussions that resonate with you.
- Add more information at any point! Useful details include: use cases, relevant labels, desired outcomes, and any accompanying screenshots.

As a member of the GitHub community, your participation is essential. While we can't promise that every suggestion will be implemented, we want to emphasize that your feedback is instrumental in guiding our decisions and priorities.

Thank you once again for your contribution to making GitHub even better! We're grateful for your ongoing support and collaboration in shaping the future of our platform. ⭐

**Comment by goran-w** [comment]
This issue is still there!

~~I tried reducing the saved-search name to one without spaces and just a single dash (as in the example from the Save-funtionality) and got it working temporarily, but then even that saved search stopped working.~~

Saved queries with spaces in their names will actually work, but they then need to be put in quotes, so I'd recommend against it anyway.

However, the issue with saved queries not working is totally unrelated to how their names are formatted - see my follow-up comment below!

**Comment by goran-w** [comment]
Hey, I think I found what's happening! My saved query had a search-string consisting of only two qualifiers:

> org:<my_org> topic:<my_topic>

I tried pasting this search-string explicitly, and got this tip-message:

> Some qualifiers in your query (topic, sort) are not supported when searching code. Try searching for repositories instead?

The "global" search defaults to `Code` in the left `Filter by` sidebar. If I switched `Filter by` to `Repositories`, my search string worked fine!

However, when the same search string was embedded in a `saved:` query and `Filter by` was `Code`, I instead got this (non-descriptive) warning-message (repeated twice, same as in the Original Post of this issue) :

> Possible unrecognized qualifier, searching for this term literally

If I then switched `Filter by` to `Repositories`, it worked fine - only there was NO HINT that this was the problem! (The above double-identical-warning message is very unhelpful, to say the least.)

All this aside, I find the `topic:` search qualifier a bit disappointing - it's too restrictive in the current implementation, as apparently it ONLY works when `Filter by` is set to `Repositories` or `Packages`. This tip-message is even a bit amusing:

> The topic qualifier is not supported when searching topics.

Ideally, I'd have expected a Code search query like the following to work, and yield results of each matching file-pattern `<my_file_pattern>` from each repo in specified organization `<my_org>` where the repo has the specified topic `<my_topic>` :

> org:<my_org> topic:<my_topic> path:<my_file_pattern>

(Question: How else can I limit a code search to just my team's repos within a larger organization?)
